src/pudl/output/ferc1.py

```diff
@@ -1063,6 +1077,88 @@ def calculations(self):
         ].copy()
         return calc_explode
 
+    @property
+    def calculations(self: Self):
+        """Remove any calculation components that aren't relevant to the explosion.
+
+        At the end of this process several things should be true:
+
+        - Only parents with table_name in the explosion tables should be retained.
+        - All calculations in which *any* components were outside of the tables in
+          the explosion should be turned into leaves -- i.e. they should be replaced
+          with a single calculation component filled with NA values.
+        - Every remaining calculation component must also appear as a parent (if it
+          is a leaf, then it will have a single null calculation component)
+        - There should be no records where only one of table_name or xbrl_factoid
+          are null. They should either both or neither be null.
+        - table_name_parent and xbrl_factoid_parent should be non-null.
+        """
```

Comment:
@cmgosnell this is a new version of the exploded calculation components that's trying to work with the new association table format, but several of the sanity checks are failing and I could use some external 👀

Reply:
i think i found at least one of your errors re my comment below